truffle/src/main/java/org/jruby/truffle/nodes/core/KernelNodes.java

```diff
@@ -1562,7 +1567,12 @@ public boolean doesRespondTo(VirtualFrame frame, Object object, RubyString name,
 
         @Specialization
         public boolean doesRespondTo(VirtualFrame frame, Object object, RubySymbol name, UndefinedPlaceholder checkVisibility) {
-            return dispatch.doesRespondTo(frame, name, object);
+            return doesRespondTo(frame, object, name, false);
+        }
+
+        @Specialization
+        public boolean doesRespondTo(VirtualFrame frame, Object object, RubySymbol name, RubyBasicObject checkVisibility) {
+            return doesRespondTo(frame, object, name, false);
```

Comment:
This is a case where a one-off `@CreateCast` could keep the number of specializations down to 1, at the cost of having to box the boolean.